src/SFA.DAS.Commitments.Database/AdhocScripts/Support/ApprenticeshipStopFixUpScript.sql

```diff
@@ -0,0 +1,184 @@
+﻿/*
+Apprenticeship Stop fix-up script
+This script stops a specific apprenticeship as at a given date. If the apprenticeship is already stopped, the stop date will be amended accordingly.
+IMPORTANT: Also generates a SQL statement to be run against the Events DB to keep things in sync.
+*/
+  
+--Set the target record here:
+declare @ApprenticeshipId bigint = 0 -- This is the specific apprenticeship record you want to stop or correct
+declare @Uln varchar(max) = 'xxxxxxxxxx' -- For extra protection against errors, also specify the ULN for the apprenticeship
+declare @StopDate DATETIME = 'yyyy-mm-dd' /* YYYY-MM-DD */ --This is the date you want to stop the apprenticeship (or correct to)
+
+/* ========================================================================== */
+/* =================== DO NOT MODIFY BELOW THIS LINE ======================== */
+/* ========================================================================== */
+   
+BEGIN TRAN
+   
+    --Just some vars here
+    DECLARE @error INT
+	declare @originalApprenticeId BIGINT
+	declare @originalUln VARCHAR(10)
+    declare @originalPaymentStatus BIGINT
+    declare @originalStartDate DATETIME
+    declare @originalStopDate DATETIME
+    declare @originalHistoryId BIGINT
+	declare @originalHistoryJson VARCHAR(MAX)
```

Comment:
OriginalState and UpdatedState in the History table are both *n*varchar(MAX), so it would be safer if @originalHistoryJson and @historyJson were the same.